### ISSUE TITLE
Upcoming forced update to Docker 25.x in AL2

### ISSUE THREAD

**Issue by achave11-ucsc** [DataBiosphere/azul#6326]
> We are contacting you about an upcoming change to the Docker version 20.10 used in Amazon Linux 2 (AL2). This version of Docker reached its end-of-life in December 2023, and will not receive any further security updates. If your service does not depend on the Docker package on AL2, you can disregard this message.
> 
> On June 12, 2024, we will update the current version of Docker to 25.0.3 from 20.10.25. A security advisory will not be applied to this initial release of Docker 25.0.3. This means running 'yum update —security' at first boot will not automatically update your EC2 instance. Newly launched EC2 instances that install Docker from the Amazon Linux RPM repositories will get Docker 25.0.3. If you cannot update the AMI you use for your EC2 instances, you can run 'sudo yum update docker' to update to Docker 25.0.3.
> 
> On August 28, 2024, we will be adding a security advisory to Docker 25.0.3. This means that existing EC2 instances that run 'yum update —security' will install Docker 25.0.3.
> 
> If you do not want to upgrade to Docker 25.0.3, we recommend locking your Docker package to the version you are currently using prior to August 28, 2024.
> 
> If you have questions, or need assistance with the upgrade, please contact AWS Support [1].
> 
> [1] https://aws.amazon.com/support

**Comment by achave11-ucsc** [comment]
Assignee to pin Docker version to whatever we are currently using, as in
```diff
Index: terraform/gitlab/gitlab.tf.json.template.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/terraform/gitlab/gitlab.tf.json.template.py b/terraform/gitlab/gitlab.tf.json.template.py
--- a/terraform/gitlab/gitlab.tf.json.template.py	(revision 7bd1543b87ef277fe8a437673cb9a9faf6297cec)
+++ b/terraform/gitlab/gitlab.tf.json.template.py	(date 1718052849233)
@@ -1642,7 +1642,7 @@
                         ['/dev/nvme1n1', gitlab_mount, 'ext4', '']
                     ],
                     'packages': [
-                        'docker',
+                        'docker@20.0.x',
                         'amazon-cloudwatch-agent',
                         'amazon-ecr-credential-helper',
                         'dracut-fips'

```

**Comment by achave11-ucsc** [comment]
Since we've somewhat frequently scrap the GitLab instance, and use the latest available AMI (#6082), it seems like the instance is already using the Docker version AWS plans to update to (25.0.3). Additionally, it appears like we do use the Amazon Linux RPM repository to install Docker (see default repositories enabled, https://docs.aws.amazon.com/linux/al2/ug/add-repositories.htm & repo info for the Docker package below). The notice reads …

> Newly launched EC2 instances that install Docker from the Amazon Linux RPM repositories will get Docker 25.0.3. If you cannot update the AMI you use for your EC2 instances, you can run 'sudo yum update docker' to update to Docker 25.0.3.

 GitLab `dev`:
 ```
 [ec2-user@ip-172-21-0-99 ~]$ yum info docker
Loaded plugins: extras_suggestions, langpacks, priorities, update-motd
Installed Packages
Name        : docker
Arch        : x86_64
Version     : 25.0.3
Release     : 1.amzn2.0.1
Size        : 167 M
Repo        : installed
From repo   : amzn2extra-docker
Summary     : Automates deployment of containerized applications
URL         : http://www.docker.com
License     : ASL 2.0 and MIT and BSD and MPLv2.0 and WTFPL
Description : Docker is an open-source engine that automates the deployment of any
            : application as a lightweight, portable, self-sufficient container that will
            : run virtually anywhere.
            : 
            : Docker containers can encapsulate any payload, and will run consistently on
            : and between virtually any server. The same container that a developer builds
            : and tests on a laptop will run at scale, in production*, on VMs, bare-metal
            : servers, OpenStack clusters, public instances, or combinations of the above.

[ec2-user@ip-172-21-0-99 ~]$ docker --version
Docker version 25.0.3, build 4debf41

```
GitLab `anvildev`:
```
[ec2-user@ip-172-23-0-207 ~]$ yum info docker
Loaded plugins: extras_suggestions, langpacks, priorities, update-motd
Installed Packages
Name        : docker
Arch        : x86_64
Version     : 25.0.3
Release     : 1.amzn2.0.1
Size        : 167 M
Repo        : installed
From repo   : amzn2extra-docker
Summary     : Automates deployment of containerized applications
URL         : http://www.docker.com
License     : ASL 2.0 and MIT and BSD and MPLv2.0 and WTFPL
Description : Docker is an open-source engine that automates the deployment of any
            : application as a lightweight, portable, self-sufficient container that will
            : run virtually anywhere.
            : 
            : Docker containers can encapsulate any payload, and will run consistently on
            : and between virtually any server. The same container that a developer builds
            : and tests on a laptop will run at scale, in production*, on VMs, bare-metal
            : servers, OpenStack clusters, public instances, or combinations of the above.

[ec2-user@ip-172-23-0-207 ~]$ docker --version
Docker version 25.0.3, build 4debf41

```

**Comment by dsotirho-ucsc** [comment]
@hannes-ucsc: "Looks like the update was sprung on us when we updated the AMI as Abraham describes above. Nothing broke so we should be good."
